docs/metrics.md

```diff
@@ -4,30 +4,173 @@ Let's look at the various metrics you can calculate using Valor.
+| AP Averaged Over IoUs | The average of several AP metrics, calculated at various IoUs, grouped by class. | $$\dfrac{1}{\|thresholds\|} \sum\limits_{iou \in thresholds} AP_{iou}$$ |
+| Mean Average Precision (mAP) 	| The mean of several AP scores, calculated over various classes. | $$\dfrac{1}{\|classes\|} \sum\limits_{c \in classes} AP_{c}$$ |
+| mAP Averaged Over IoUs | The mean of several averaged AP scores, calculated over various classes. | $$\dfrac{1}{\|thresholds\|} \sum\limits_{iou \in thresholds} mAP_{iou}$$ |
 
 ## Semantic Segmentation Metrics
 
-| Name 	| Description 	|
-|---	|---	|
-| Intersection Over Union (IOU) 	| The overlap between the ground truth and predicted regions of an image, measured as a percentage, grouped by class. IOUs are calculated by a) fetching the ground truth and prediction rasters for a particular image and class, b) counting the true positive pixels (e.g., the number of pixels that were selected in both the ground truth masks and prediction masks), and c) dividing the sum of true positives by the total number of pixels in both the ground truth and prediction masks. |
-| Mean IOU 	| The average of IOUs, calculated over several different classes. 	|
+| Name | Description | Equation |
+| :- | :- | :- |
+| Intersection Over Union (IoU) | A ratio between the groundtruth and predicted regions of an image, measured as a percentage, grouped by class. |$$\dfrac{area( prediction \cap groundtruth )}{area( prediction \cup groundtruth )}$$ |
+| Mean IoU 	| The average of IoUs, calculated over several different classes. | $$\dfrac{1}{\|classes\|} \sum\limits_{c \in classes} IoU_{c}$$ |
+
+# Methods
+
+## ROC AUC for Classification
+
+### Determining Binary Truth
+
+| Element | Description |
+| ------- | ------------ |
+| True Positive (TP) | Prediction confidence is above a threshold and is correct. |
+| False Positive (FP) | Prediction confidence is above a threshold and is incorrect. |
+| True Negative (TN) | Prediction confidence is below a threshold and is correct. |
+| False Negative (FN) | Prediction confidence is below a threshold and is incorrect. |
+
+- $\text{True Positive Rate (TPR)} = \dfrac{|TP|}{|TP| + |FN|}$
+
+- $\text{False Positive Rate (FPR)} = \dfrac{|FP|}{|FP| + |TN|}$
+
+### Receiver Operating Characteristic (ROC)
+
+An ROC curve plots TPR vs. FPR at different confidence thresholds.
+
+### Area under the ROC curve (ROC AUC)
+
+From the ROC curve we calculate the ROC AUC metric by taking the integral using the trapezoidal rule formula.
+
+$$
+\sum_{i=1}^{|scores|} \frac{(TPR(scores_{i-1}) - TPR(scores_i)) \cdot (FPR(scores_{i-1}) - FPR(scores_i))}{2}
+$$
+
+### References
+- [Classification: ROC Curve and AUC](https://developers.google.com/machine-learning/crash-course/classification/roc-and-auc)
```

Comment:
nit: can we delete this reference section since there's only one reference, and hyperlink to this page somewhere else in your above explaination?

Reply:
i agree if there's just one link in reference section than the section is unnecessary